.github/workflows/ci.yml

```diff
@@ -1,146 +1,151 @@
-  STANFORD_MODELS: /home/runner/third/stanford-postagger
-  STANFORD_POSTAGGER: /home/runner/third/stanford-postagger
-  SENNA: /home/runner/third/senna
-  PROVER9: /home/runner/third/prover9/bin
-  MEGAM: /home/runner/third/megam
-  # TADM requires `libtaopetsc.so` from PETSc v2.3.3, and likely has more
-  # tricky to install requirements, so we don't run tests for it.
-  # TADM: /home/runner/third/tadm/bin
-  MALT_PARSER: /home/runner/third/maltparser
+  THIRD_PARTY_DIR: ${{ github.workspace }}/third
+  CORENLP: ${{ github.workspace }}/third/stanford-corenlp
+  CORENLP_MODELS: ${{ github.workspace }}/third/stanford-corenlp
+  STANFORD_PARSER: ${{ github.workspace }}/third/stanford-parser
+  STANFORD_MODELS: ${{ github.workspace }}/third/stanford-postagger
+  STANFORD_POSTAGGER: ${{ github.workspace }}/third/stanford-postagger
+  SENNA: ${{ github.workspace }}/third/senna
+  PROVER9: ${{ github.workspace }}/third/prover9/bin
+  MEGAM: ${{ github.workspace }}/third/megam
+  MALT_PARSER: ${{ github.workspace }}/third/maltparser
 
 jobs:
   pre-commit:
-    name: Run pre-commit
+    name: pre-commit
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v5
-      - uses: actions/setup-python@v6
+
+      - name: Set up Python
+        uses: actions/setup-python@v6
         with:
-          python-version: '3.12'
-      - run: |
+          python-version: "3.12"
+
+      - name: Install pre-commit
+        run: |
           pip install pre-commit
           pre-commit run --all-files
 
-  cache_nltk_data:
-    name: Cache nltk_data
-    needs: pre-commit
+  minimal_download_test:
+    name: Minimal NLTK Download Test
+    runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        os: [ubuntu-latest, windows-latest]
-    runs-on: ${{ matrix.os }}
+        os: [ubuntu-latest, macos-latest, windows-latest]
     steps:
-      - name: Checkout code
-        uses: actions/checkout@v5
+      - uses: actions/checkout@v5
 
-      - name: Cache nltk data
-        uses: actions/cache@v4
-        id: restore-cache
+      - name: Set up Python
+        uses: actions/setup-python@v6
         with:
-          path: ~/nltk_data
-          key: nltk_data_${{ secrets.CACHE_VERSION }}
-
-      - name: Download nltk data packages on cache miss
-        run: |
-          pip install regex # dependencies needed to download nltk data
-          python -c "import nltk; from pathlib import Path; path = Path('~/nltk_data').expanduser(); path.mkdir(exist_ok=True); nltk.download('all', download_dir=path)"
-        shell: bash
-        if: steps.restore-cache.outputs.cache-hit != 'true'
+          python-version: "3.12"
 
-  cache_third_party:
-    name: Cache third party tools
-    needs: pre-commit
-    runs-on: ubuntu-latest
-    steps:
-      - name: Checkout code
-        uses: actions/checkout@v5
+      - name: Install NLTK and dependencies
+        run: pip install --upgrade pip && pip install --upgrade nltk
```

Comment:
We are installing nltk from pypi and importing it: https://github.com/nltk/nltk/pull/3427/files#diff-b803fcb7f17ed9235f1e5cb1fcd2f5d3b2838429d4368ae4c57ce4436577f03fR63

The download test runs the pypi version and not the version from the git repo if I am not mistaken.